### PR TITLE
[Enhancement/Test] Add support for Pytorch JIT traced/scripted models

### PR DESCRIPTION
**Pull request by telescopic**
## Description

- Add support for Pytorch JIT traced/scripted modules.
- Standard `cloudpickle` does not support serialization of TorchScript modules, instead `torch.jit.save()` and `torch.jit.load()` are used

## Motivation and Context

Currently, PytorchModelArtifact does not support scripted/traced models since `cloudpickle` cannot serialize ScriptModules

Fixes issue #873 

## How Has This Been Tested?

Added integration tests for saving/loading scripted and traced models.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [x] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [x] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [x] My change reduces project test coverage and requires unit tests to be added
- [x] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by telescopic** [comment]
@flosincapite / @yubozhao / @parano Please review my PR ! 
Thanks in advance :D

**Comment by yubozhao** [comment]
Looks great.  Good job @telescopic!

File: bentoml/frameworks/pytorch.py [hunk 1]
```diff
@@ -75,7 +76,8 @@ def pack(self, model, metadata=None):  # pylint:disable=arguments-differ
 
         if not isinstance(model, torch.nn.Module):
             raise InvalidArgument(
-                "PytorchModelArtifact can only pack type 'torch.nn.Module'"
+                "PytorchModelArtifact can only pack type \
```
Comment (parano) [inline bentoml/frameworks/pytorch.py:79]: Could you add a usage example in the docstring above in this class?

Comment (telescopic) [reply bentoml/frameworks/pytorch.py:116]: Thanks for the suggestion :D
Updated accordingly!

**Comment by parano** [comment]
Thanks again @telescopic! Merging now!

**Pull request merged**